internal/scanner/scanner.go

```diff
@@ -2341,11 +2341,11 @@ func ComputePositionOfLineAndCharacter(lineStarts []core.TextPos, line int, char
 	return res
 }
 
-func GetLeadingCommentRanges(f *ast.NodeFactory, text string, pos int) iter.Seq[ast.CommentRange] {
+func GetLeadingCommentRanges(f ast.NodeFactoryCoercible, text string, pos int) iter.Seq[ast.CommentRange] {
```

Comment:
It seems like these could be concrete, right? I think most of the time we'll have `*NodeFactory` when calling these pair.

Reply:
That's probably fair for this case. There are only four places in the printer where we would have to write `p.emitContext.Factory.AsNodeFactory()` as the transforms don't generally depend on comments in the source.